paddle/phi/common/data_type.h

```diff
@@ -153,6 +153,8 @@ PD_FOR_EACH_DATA_TYPE(PD_SPECIALIZE_CppTypeToDataType)
 
 #undef PD_SPECIALIZE_CppTypeToDataType
 
+#undef PD_FOR_EACH_DATA_TYPE
+
```

Comment:
After the macro PD_FOR_EACH_DATA_TYPE has accomplished its mission,  it seems  ok  to  #undef  it.